config/management/src/smoke_test.rs

```diff
@@ -0,0 +1,167 @@
+        network.discovery_method = DiscoveryMethod::Onchain;
+        config.full_node_networks = vec![NetworkConfig::default()];
+        config.randomize_ports();
+
+        let validator_network = config.validator_network.as_mut().unwrap();
+        let validator_network_address = validator_network.listen_address.clone();
+        let identity_key = storage
+            .export_private_key(libra_global_constants::VALIDATOR_NETWORK_KEY)
+            .unwrap();
+        let identity_key: x25519::PrivateKey = identity_key.to_bytes().as_ref().try_into().unwrap();
+        // This is a deprecated field
+        let signing_key = Ed25519PrivateKey::generate_for_testing();
+        validator_network.network_keypairs = Some(NetworkKeyPairs::load(identity_key, signing_key));
+
+        let fullnode_network = &mut config.full_node_networks[0];
+        let identity_key = storage
+            .export_private_key(libra_global_constants::FULLNODE_NETWORK_KEY)
+            .unwrap();
+        let identity_key = identity_key.to_bytes().as_ref().try_into().unwrap();
+        // This is a deprecated field
+        let signing_key = Ed25519PrivateKey::generate_for_testing();
+        fullnode_network.network_keypairs = Some(NetworkKeyPairs::load(identity_key, signing_key));
+        let fullnode_network_address = fullnode_network.listen_address.clone();
+
+        configs.push(config);
+
+        helper.operator_key(&ns, &ns_shared).unwrap();
+        helper
+            .validator_config(
+                AccountAddress::random(),
+                validator_network_address,
+                fullnode_network_address,
+                &ns,
+                &ns_shared,
+            )
+            .unwrap();
+    }
+
+    // Step 4) Produce genesis and introduce into node configs
+    let genesis = helper.genesis().unwrap();
+
+    // Step 5) Produce a waypoint
+    // TODO(davidiw) Push this logic into this tool and not this test
+    let path = TempPath::new();
+    let db_rw = DbReaderWriter::new(LibraDB::open(&path, false, None).unwrap());
+    let waypoint = db_bootstrapper::bootstrap_db_if_empty::<LibraVM>(&db_rw, &genesis)
+        .unwrap()
+        .unwrap();
+
+    // Step 5) Introduce waypoint and genesis into the configs
+    let temppath = TempPath::new();
+    temppath.create_as_dir().unwrap();
+    let swarm_path = temppath.path().to_path_buf();
+
+    for (i, mut config) in configs.iter_mut().enumerate() {
+        let waypoint_value = Value::String(waypoint.to_string());
+        helper
+            .storage(i.to_string())
+            .set(libra_global_constants::WAYPOINT, waypoint_value)
+            .unwrap();
+
+        let mut to = swarm_path.clone();
+        to.push("secure_store_for_".to_string() + &i.to_string());
+        std::fs::copy(helper.path_string(), &to).unwrap();
+
+        let mut storage_config = OnDiskStorageConfig::default();
+        storage_config.path = to;
+        storage_config.set_data_dir(PathBuf::from(""));
```

Comment:
is this not used?

Reply:
it defaults to something, but no, it is not, if the path is a full path, this is ignored, but I want to be careful to not leave some unused data here.